### PR TITLE
feat: add Jws2020 cryptosuite

### LINKED ISSUE #482: SSI: Add support for JsonWebSignature2020
## WHAT

The SSI solution that currently being implemented in Tractus-X mandates the use of the [JsonWebSignature2020](https://www.w3.org/community/reports/credentials/CG-FINAL-lds-jws2020-20220721) crypto suite. 

## WHY
Jws2020 is required in Tractus-/Catena-X and it should be aligned with architectural and code-level [decisions](https://github.com/eclipse-tractusx/tractusx-edc/tree/main/docs/development/decision-records/2023-06-13_use_of_iron_library) made in Tractus-X/EDC.

## FURTHER NOTES

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## WHAT

Adds the [JsonWebSignature2020](https://www.w3.org/community/reports/credentials/CG-FINAL-lds-jws2020-20220721) crypto suite for use with VerifiableCredentials and VerifiablePresentations.  

## WHY

Catena-X mandates the use of JsonWebSignature2020 for VerifiableCredentials and VerifiablePresentations, so we'll need to be able to verify proofs created with that cryptosuite.

## Known limitations

- verifying VC/VP that have been signed with the `did:key` verification method is not supported due to [implementation feasibility concerns](https://w3c-ccg.github.io/did-method-key/#signature-method-creation-algorithm) and the fact that the [Multibase java impl](https://w3c-ccg.github.io/did-method-key/#signature-method-creation-algorithm), that iron relies on, is not available on MavenCentral
- signing VC/VP with the `did:key` method is possible, but only if the `did:key` is generated out-of-band
- `OctetKeyPair`s are only supported, if the [tink](https://mvnrepository.com/artifact/com.google.crypto.tink/tink) library is present at runtime. They arent' supported by the Java17 JCA yet.
- switched Checkstyle verbosity to default `true`

Closes #482

### CODE REVIEW

File: edc-extensions/ssi/jws2020-crypto-suite/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+VerifiableCredentials and VerifiablePresentations are processed as JSON(-LD) objects, so some familiarity with JSON-LD
+is required.
+The entrypoint into the cryptographic suite is the `Vc` class, which allows signing/issuing and verifying JSON-LD
+structures. The following samples use explicit types for clarity. These are just some illustrative examples, please
+check the `IssuerTests` and the `VerifierTests` for more comprehensive explanations.
+
+### Sign a VC
+
+```java
```
Comment (ndr-brt) [inline edc-extensions/ssi/jws2020-crypto-suite/README.md:21]: could be this code snipped formatted correctly?

File: edc-extensions/ssi/jws2020-crypto-suite/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ */
+plugins {
+    `java-library`
+}
+
+dependencies {
+    api(libs.edc.spi.jwt)
+    implementation(libs.nimbus.jwt)
+    implementation(libs.edc.spi.jsonld)
```
Comment (ndr-brt) [inline edc-extensions/ssi/jws2020-crypto-suite/build.gradle.kts:21]: nit: `api`

File: edc-extensions/ssi/jws2020-crypto-suite/src/main/java/org/eclipse/edc/security/signature/jws2020/JwkMethod.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Override
+    public byte[] publicKey() {
+        return keyPair != null ? serializeKeyPair(keyPair.toPublicJWK()) : null;
+    }
+
+    private byte[] serializeKeyPair(JWK keyPair) {
+        try {
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            ObjectOutputStream out = new ObjectOutputStream(bos);
```
Comment (ndr-brt) [inline edc-extensions/ssi/jws2020-crypto-suite/src/main/java/org/eclipse/edc/security/signature/jws2020/JwkMethod.java:41]: `var`

**Review by ndr-brt — APPROVED** [review]

File: edc-extensions/ssi/jws2020-crypto-suite/README.md (hunk 1 above)
Comment (paullatzelsperger) [reply edc-extensions/ssi/jws2020-crypto-suite/README.md:21]: jeez, IDK whatever ungodly code formatter mangled this... 😱

**Review by wolf4ood — APPROVED** [review]
:rocket:

**Pull request merged**